File: LeetcodeEasyProblems/_202. Happy Number/Solution.java
import java.util.HashSet;
class Solution {
    public boolean isHappy(int n) {
        // Create a visited hash map to 
        // keep track of each visited number
        HashSet<Integer> visitiedArr = new HashSet<>();
        while (!visitiedArr.contains(n)){
            // Add the number to visited array
            visitiedArr.add(n);
            // Update n into the sum of its
            // digit squared
            n = this.sumSquareDigit(n);
        }
        // After the loop, n become a value 
        // that's visited
        if (n == 1){
            return true;
        }
        return false;
        
    }
    public int sumSquareDigit(int number){
        // Initialize the sum of its digit squared
        int sum = 0;
        while (number != 0){
            // Add the digit squared to the sum
            sum += (number%10) * (number%10);
            // Update the number to the one without 
            // the digit
            number /= 10;
        }

        return sum;
    }
}

File: LeetcodeEasyProblems/_202. Happy Number/Solution.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        # Create a hashset to keep track of 
        # all visited number
        visit = set()
        # The loop runs until there's a duplicate
        while n not in visit:
            # Add the number into the hashset
            visit.add(n)
            # Update the number to the sum
            # of its digit squared
            n = self.squareOfDigit(n)
        # After the loop, if n == 1, return True
        if n == 1:
            return True
        # Else, return False
        return False
    
    def squareOfDigit(self, number):
        # Initialize the sum of the digit squared
        sum = 0
        # The loop runs until the number == 0
        while number:
            sum +=  pow((number%10),2)
            # After adding the digit squared, update
            # the number
            number //= 10
        return sum


        